### PR TITLE
[config] Rework some APIs

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- Reworked some config and denylist APIs to make it more clear what values are being read
- For deny list, this manifested in having "next_epoch" and "current_epoch" APIs

## Test plan 

- New tests 

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-framework/packages/sui-framework/sources/coin.move [hunk 1]
```diff
@@ -315,13 +315,21 @@ module sui::coin {
...
         deny_list: &DenyList,
         addr: address,
         ctx: &TxContext,
     ): bool {
         let ty = type_name::get_with_original_ids<T>().into_string().into_bytes();
-        deny_list.v2_most_recent_contains(DENY_LIST_COIN_INDEX, ty, addr, ctx)
+        deny_list.v2_contains_current_epoch(DENY_LIST_COIN_INDEX, ty, addr, ctx)
+    }
+
+    public(package) fun deny_list_v2_contains_next_epoch<T>(
```
Comment (tzakian) [inline crates/sui-framework/packages/sui-framework/sources/coin.move:327]: the lining up of this function's name + the `deny_list` name + receiver syntax + the other method name is just perfection.

File: crates/sui-framework/packages/sui-framework/sources/config.move [hunk 2]
```diff
@@ -181,18 +185,18 @@ module sui::config {
         data.newer_value.borrow_mut()
     }
 
-    public(package) fun borrow_most_recent<
+    public(package) fun read_newer_setting<
```
Comment (tzakian) [inline crates/sui-framework/packages/sui-framework/sources/config.move:188]: 100% fine with this name (and apologies for possibly relitigating the name...) but ... what do you think of `read_setting_next_epoch` as a possible other name for this and that kinda fits in with the other names a bit more uniformly 🫣 😬

**Review by tzakian — APPROVED** [review]
Changes look good to me. Just one small thing on naming that you should feel free to ignore if you want :)

File: crates/sui-framework/packages/sui-framework/sources/config.move (hunk 2 above)
Comment (tnowacki) [reply crates/sui-framework/packages/sui-framework/sources/config.move]: Switched everything to `for_next_epoch`

Comment (tnowacki) [reply crates/sui-framework/packages/sui-framework/sources/config.move]: Well, DenyList is still `current_epoch` and `next_epoch` but I think the `for_` helps in the Config side

**Pull request merged**